runtime/src/leader_bank_status.rs

```diff
@@ -0,0 +1,167 @@
+use {
+    crate::bank::Bank,
+    solana_sdk::slot_history::Slot,
+    std::{
+        sync::{Arc, Condvar, Mutex, RwLock, Weak},
+        time::{Duration, Instant},
+    },
+};
+
+/// Tracks leader status of the validator node and notifies when:
+///     1. A leader slot begins
+///     2. A leader slot completes
```

Comment:
this is a bit vague (maybe in purpose, right?). admittedly, this is in `solana-runtime`, so some generality can be desired in some cases. but, i think it helps reader in this case to be explicit for intended use case like this:

```
/// Tracks leader bank among bank_forks and notifies when:
///     1. A leader bank initiates (=poh-initiated)
///     2. A leader bank completes (=poh-completed)
```

Reply:
Updated here: bea9edccc539d3cc70b78890eab8c47618bbb0e5